lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunicationServerCommon.cpp

```diff
@@ -1304,6 +1304,9 @@ void GDBRemoteCommunicationServerCommon::
     if (!abi.empty())
       response.Printf("elf_abi:%s;", abi.c_str());
     response.Printf("ptrsize:%d;", proc_arch.GetAddressByteSize());
+    std::optional<bool> non_resumable = proc_info.IsNonResumable();
+    if (non_resumable)
+      response.Printf("non_resumable:%d", *non_resumable);
```

Comment:
This is part of qProcessInfo (https://lldb.llvm.org/resources/lldbgdbremote.html#qprocessinfo) which is, I presume, only requested once because all the information is constant for the process lifetime.

At least for the situation at hand, non-resumeable is also constant. Though the process had to get into that state somehow, but if you were debugging it before the non-resumable point, it wouldn't have got into the non-resumeable state anyway so it makes no difference.

So unless anyone can think of a situation where non-resumeable could change, this packet is probably fine.

Reply:
Your intuition is correct, for now this works, but in the future (if we want to support `O_TRACEEXIT`), we would need to update this. Currently I can get away with attach returning a process info that we can't resume. 

This is still very WIP, as I'm trying to sort with Greg the gotcha's. I will break this patch up into pieces soon :)

Reply:
Cool, so something would get added to the remote protocol to make this work but exactly what we can decide later.